File: Proxy-Server/src/main/kotlin/com/shadowell/cloud/proxyServer/dao/UserDao.java
package com.shadowell.cloud.proxyServer.dao;


import com.shadowell.cloud.proxyServer.entity.SysUser;
import org.springframework.data.jpa.repository.JpaRepository;


public interface UserDao extends JpaRepository<SysUser, Long>{

	SysUser findByUsername(String username);
}

File: Security-Server/src/main/kotlin/com/shadowell/cloud/Dao/UserDao.kt
package com.shadowell.cloud.Dao

import com.shadowell.cloud.Model.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.security.core.userdetails.UserDetails
import java.util.*

interface UserDao: JpaRepository<User, Long> {
    fun findByUsername(username: String): UserDetails ?
}